plutus-core/executables/uplc/Main.hs

```diff
@@ -160,45 +165,44 @@ runApply (ApplyOptions inputfiles ifmt outp ofmt mode) = do
 ---------------- Evaluation ----------------
 
 runEval :: EvalOptions -> IO ()
```

Comment:
This function just keeps getting more complicated with all the different branches.  I suppose there's not much to be done though, because it has to handle all the different combinations of flags.

Reply:
Yeah, I tried to make it *less* complicated, by e.g. handling the budget options once rather than repeatedly, but it's a mess.